samples/model-builder/conftest.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from unittest.mock import MagicMock, patch
+
+from google.cloud import aiplatform
+
+
+@pytest.fixture
+def mock_sdk_init():
+    with patch.object(aiplatform, "init") as mock:
+        yield mock
+
+
+# ----------------------------------------------------------------------------
+# Dataset Fixtures
+# ----------------------------------------------------------------------------
+
+
+@pytest.fixture
+def mock_init_dataset():
+    with patch.object(aiplatform.datasets.Dataset, "__init__") as mock:
+        mock.return_value = None
+        yield mock
+
+
+@pytest.fixture
+def mock_create_dataset():
+    with patch.object(aiplatform.datasets.Dataset, "create") as mock:
+        mock.return_value = MagicMock(aiplatform.Dataset)
+        yield mock
+
+
+@pytest.fixture
+def mock_create_image_dataset():
+    with patch.object(aiplatform.datasets.ImageDataset, "create") as mock:
+        mock.return_value = MagicMock(aiplatform.Dataset)
+        yield mock
+
+
+# ----------------------------------------------------------------------------
+# TrainingJob Fixtures
+# ----------------------------------------------------------------------------
+
+
+@pytest.fixture
+def mock_init_automl_image_training_job():
+    with patch.object(
+        aiplatform.training_jobs.AutoMLImageTrainingJob, "__init__"
+    ) as mock:
+        mock.return_value = None
```

Comment:
This seems either superfluous or possibly will cause an error because we would call `run` on `None`.

Reply:
Not returning `None` when patching the `__init__` caused the following exception:
```
TypeError: __init__() should return None, not 'MagicMock'
```

The downstream calls would also be patched so it shouldn't cause an error. Is this alright or should we look for alternative approaches?

Reply:
Ah, right. I was wrong on that. Leave it as is. That makes sense, `__new__` returns the instance, not `__init__`.